src/network-device.js

```diff
@@ -13,6 +16,59 @@ const NetworkStatus = fromProtobufEnum(proto.NetworkState, {
 	UP: 'UP'
 });
 
+const InterfaceType = fromProtobufEnum(proto.InterfaceType, {
+	INVALID : 'INVALID_INTERFACE_TYPE',
+	LOOPBACK : 'LOOPBACK',
+	THREAD : 'THREAD',
+	ETHERNET : 'ETHERNET',
+	WIFI : 'WIFI',
+	PPP : 'PPP'
+});
+
+const InterfaceConfigurationSource = fromProtobufEnum(proto.InterfaceConfigurationSource, {
+	NONE: 'NONE',
+	DHCP: 'DHCP',
+	STATIC: 'STATIC',
+	SLAAC: 'SLAAC',
+	DHCPV6: 'DHCPV6'
+});
+
+/**
+* Converts a given interface IP address object with version into a dotted-decimal format.
+*
+* @param {object} ifaceAddr
+* @param {string} version 'v4' or 'v6'
+* @returns {string} address in dotted-decimal format
+*/
+function convertInterfaceAddress(ifaceAddr, version) {
+	let res = null;
+	let prefixLength = null;
+	if (ifaceAddr && ifaceAddr.address && ifaceAddr.address[version]) {
+		const addrObj = ifaceAddr.address[version];
+		res = convertIPAddress(addrObj, version);
+	}
+	if (ifaceAddr && ifaceAddr.prefixLength) {
+		prefixLength = ifaceAddr.prefixLength;
+	}
+	return res ? `${res}/${prefixLength}` : null;
+}
+
+/**
+* Converts a given IP address object with version into a dotted-decimal address string.
+*
+* @param {object} addr addr.address sent as int32 for 'v4' and as a buffer for 'v6'
+* @param {string} version 'v4' or 'v6'
+* @returns {string} address in dotted-decimal format
+*/
+function convertIPAddress(addr, version) {
+	if (addr && addr.address) {
+		const val = addr.address;
+		const res = version === 'v4' ? Address4.fromInteger(val) : Address6.fromByteArray(val);
+		return res.address;
+	}
+	return null;
+}
```

Comment:
`addr` won't have an `address` subkey. That's only for the `InterfaceConfig`

```suggestion
/**
* Converts a given IP address object with version into a dotted-decimal address string.
*
* @param {object} addr addr.address sent as int32 for 'v4' and as a buffer for 'v6'
* @param {string} version 'v4' or 'v6'
* @returns {string} address in dotted-decimal format
*/
function convertIPAddress(addr, version) {
	if (!addr) {
		return addr;
	}
	 return addr.v4 ? Address4.fromInteger(addr.v4) : Address6.fromByteArray(addr.v6);
}
```

I prefer the `if (!addr)` form rather than putting the entire function in an if. That's called a guard clause and it makes it clear what are the conditions to return early. It also avoids additional nesting for the actual core code of the function.

Reply:
I think `addr` here is meant to be either `Ipv4Address` or `Ipv6Address` in parsed form. In that case, there will be an `address` property.